### ISSUE TITLE
Support for set #{} as a spec predicate/collection

### ISSUE THREAD

**Issue by Severed-Infinity** [metosin/spec-tools#67]
In `clojure.spec` you can use `#{}` as predicates in specs but using it with spec-tools I have to wrap it in `clojure.spec/spec` for it to be valid otherwise, it results in a vague error about the issue.
```
CompilerException java.lang.Exception: Unable to resolve spec: :deg, compiling:(css_spec.clj:33:14) 
```
that error is the result of the following, and any subsequent code using `#{}` directly will produce the same error
```
(s/def ::deg (st-ds/spec ::deg {:magnitude spec/int? :unit #{:deg}}))
```
the solution is fairly straight forward, just wrap in `clojure.spec/spec`
```
(s/def ::deg (st-ds/spec ::deg {:magnitude spec/int? :unit (s/spec #{:deg})}))
```
but it would be nice to have first class support for them like the rest of the collection types.

**Comment by miikka** [comment]
Sets in data specs work the same way as vectors. For example, if you do `#{int?}`, this results in a spec for a set of ints. Unfortunately this collides with the common pattern of using sets as predicates, as you have discovered.

I'm not sure what we should do to alleviate this problem, but at least we should mention it in the documentation. I bet you won't be the last person to bump on this.

(FYI: both me and Tommi are in vacation right now so it may take a while before we can get back to this.)

**Comment by Severed-Infinity** [comment]
Ah, that makes sense, for some reason I just didn't see sets as collections like vectors. 
The only solution I can think of is quite crude and that would be to some how forcibly exclude sets as a collection option but again only thinking of this off the top of my head as I write this reply.

**Comment by ikitommi** [comment]
I updated the README to contain note of this and made the error messages better when using data-specs with non-homogeneous collections.

**Comment by ikitommi** [comment]
btw, when calling `ds/spec`, all non-regex leaf nodes are automatically wrapped as spec records. So, these mean the same thing:

```clj
(ds/spec ::user {:name spec/string?, :age spec/int?})
(ds/spec ::user {:name string?, :age int?})
```

Might be a good idea to keep those explicit, but wanted to save time on typing.
